### ISSUE TITLE
[Suggestion/Ideas for gameplay] Void Pearls and Building systems and progression.  

### ISSUE THREAD

**Issue by CoasterBlock** [tsunamayo/Starship-EVO#5973]
I think there should be a change made to how we find, and harvest void pearls, and some ideas of how the building aspect could be improved for later survival gameplay. 

**VOID PEARLS**

Currently, they can be found on planets and I think that isnt good for gameplay/progression or exploration. Achieving FTL should be something you feel proud about. 
I have two ideas: 

Idea 1:

Warp pearls are found only in specific locations, such as asteriod belts close to stars, and on the edge of the solar system. 
Alternatively around certain gas-giants. 

Idea 2: Void pearls arent found it in the world, they need to be made. Specifially, the player would need to build a structure around a black hole to harvest some kind of resource. Lets say "Dark-energy". This resource could either be pumped directly into the Warpcore or it would need to be further refined into void pearls. That way there would be an actual need to setup industry and build something large. 

-----------------------------------------------------------------------------------------------------------------------------------------

**BUILDING**

This brings me to the building progression. I think the player should need to build large structures to make use of the building system we have. Eventually, the player should be able to harvest fuel directly from stars using a special "tractor beam" for example. So a space station close to a planet would be needed. The "tractor beam" brick though would have a *minimum* size of 4m. And it would need equally as much energy. That would rewards long-term planning and make sure players actually "claim" systems and build "megastructures" and buildings in general with some kind of use. 

Bricks would also come with a techtier. So higher tier reactors are multiblocks for example. You need to make structure out of ordinary bricks plus functions. So input/output bricks, a "controller" etc. (Example from modded minecraft) The higher tier bricks should be able larger by default. So a tier 2 reactor would be 2m or 4m by default. While a tier 3 reactor would be a multiblock construction. This would encourage an engineering and design aspect of the game, but it wouldnt force people to build their ships too differently as the multiblocks could be shaped in any form, as long as they meet certain requirements. A certain volume for example. Multiblocks really are a great workaround this whole "spam 100000 reactorsblocks in an area and pipe them all up" It would allow players to make much more compact and much higher efficiency systems. 

Mechanical blocks should scale in speed. A 16m piston, should travel 16m/s. Same goes for rails, 16m rails should travel at 16m/s by default. While smaller mechanisms should travel slower. so a 1m piston, travels at 1m/s. Currently all mechanism have the same base speed regardless of scale. Which is nonsensical, and also "Punishes" larger structures. 

Suggestion to add a "ghost piston". A piston that does NOT have mechnical link between its base and platform. Would be crafted using higher tier resource and would need power to move. 

I heavily suggest looking at how modded-minecraft makes their buidlings. Create, GregTech, NuclearCraft, Instruial Engineering etc. 
It doesnt mean the systems need to be as complex in the beginning, but with higher tech, some complexity is rewarding, plus there can be an in-game "Wiki" with 3-d models that show examples on how to make stuff. Plus the workshop would allow people to install pre-built system-blueprints if they dont feel like going through the "engineering" part of the game. 

-----------------------------------------------------------------------------------------------------------------------------------------

**TRANSPORT**

In the later game, the player would be able to automate things. The player could build drone ships to fly between systems and dock at stations to transport fuel/energy/goods around their "empire" 
In the late-midgame some kind of "Portal" or "teleporter" should be available. Teleporters could be used only on the same entity. So you can teleport to different parts of a station or ship, but not to a different station or ship. Portals on the other hand, could transport across ships and maybe into other solar systems. They would also be able to transfer entities if built large enough. 

These are just some ideas, feedback is apreciated. :) 

Long story short, there should be much more incentive to build lagre, and complex systems to make space-exploration and ship/station building actually useful beyond just a fun building game. There could also be achievents for certain technological or explorative milestones. This way it would be interesting to see what is in the galaxy, and make use of the incredibly well done building system. Eventually, ships or stations made using 8m/16m/32m should be something you NEED.

**Comment by boxfoxscoot** [comment]
a separate type of rotor/rotator block that is infinitely rotating would be really nice I think there were performance concerns but if people can haphazardly glue rotors together to get it working then I think a block that rotates infinitely would be fine and also would be a lot more polished than the aforementioned buggy mess of rotors

**Comment by Majikmonster** [comment]
> a separate type of rotor/rotator block that is infinitely rotating would be really nice I think there were performance concerns but if people can haphazardly glue rotors together to get it working then I think a block that rotates infinitely would be fine and also would be a lot more polished than the aforementioned buggy mess of rotors

This. Building rotor and hinge chains that flip to achieve continuous rotation takes at least 4 mechanisms, logic to operate, and extra space to move internally. If it were a setting in the existing rotor itself or a separate brick, it would be a huge relief to those of us who don't want to keep stacking mechanisms and carving out pockets of wiggle room for a simple spin, and make the feature more accessible to less mechanically inclined players.

**Comment by boxfoxscoot** [comment]
> > a separate type of rotor/rotator block that is infinitely rotating would be really nice I think there were performance concerns but if people can haphazardly glue rotors together to get it working then I think a block that rotates infinitely would be fine and also would be a lot more polished than the aforementioned buggy mess of rotors
> 
> This. Building rotor and hinge chains that flip to achieve continuous rotation takes at least 4 mechanisms, logic to operate, and extra space to move internally. If it were a setting in the existing rotor itself or a separate brick, it would be a huge relief to those of us who don't want to keep stacking mechanisms and carving out pockets of wiggle room for a simple spin, and make the feature more accessible to less mechanically inclined players.

also more stable as theres no solution that doesnt break the moment the game lags
